### PR TITLE
ref(ui): Remove SearchInput component in favor of SearchBar

### PR DESCRIPTION
**Pull request by matejminar**
This PR removes SearchInput in favor of SearchBar which is pretty much the same thing and more widely used.

I created SearchInput for some reason during my very first days at Sentry - I probably didn't know about SearchBar.

I am killing it so that we don't have multiple components that do the same thing.
I had to add two new props to SearchBar so that it can be used in place of SearchInput (`width` and `onChange`).

We still should refactor SearchBar to use components instead of classNames in the future, but this is a step in the right direction.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/searchBar.tsx [hunk 1]
```diff
@@ -104,4 +114,8 @@ class SearchBar extends React.PureComponent<Props, State> {
   }
 }
 
+const Input = styled('input')`
+  width: ${p => (p.width ? p.width : undefined)};
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/searchBar.tsx:118]: Will this result in `width: undefined`? Perhaps this whole property should be conditional?

**Review by markstory — COMMENTED** [review]
Looks pretty good there are some alignment diffs in issue details that you might want to look at.

File: src/sentry/static/sentry/app/components/searchBar.tsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/components/searchBar.tsx:118]: Emotion seems to be smart enough to omit the thing if the returned value is undefined.
```
const Input = styled('input')`
  width: ${p => p.width ?? undefined};
  /* same result */
  ${p => p.width && `width: ${p.width}`};
`;
```

Note that if you would type undefined explicitly like this, width would be 'undefined' in the final css:
```
const Input = styled('input')`
  width: undefined;
`;
```

**Review by markstory — APPROVED** [review]
Code and percy diffs look good to me.

**Pull request merged**